haystack/pipelines/base.py

```diff
@@ -1446,3 +1395,257 @@ def __call__(self, *args, **kwargs):
+            line_prefix = "# " if mod == MODULE_NOT_FOUND else ""
+            code_lines.append(f"{line_prefix}from {mod} import {import_types}")
+
+        code = "\n".join(code_lines)
+        return code
+
+    @classmethod
+    def _order_components(
+        cls, dependency_map: Dict[str, List[str]], components_to_order: Optional[List[str]] = None
+    ) -> List[str]:
+        ordered_components = []
+        if components_to_order is None:
+            components_to_order = list(dependency_map.keys())
+        for component in components_to_order:
+            dependencies = dependency_map[component]
+            ordered_dependencies = cls._order_components(
+                dependency_map=dependency_map, components_to_order=dependencies
+            )
+            ordered_components += [d for d in ordered_dependencies if d not in ordered_components]
+            if component not in ordered_components:
+                ordered_components.append(component)
+        return ordered_components
+
+
+class _PipelineEvalReportGen:
+    @classmethod
+    def print_eval_report(
+        cls,
+        eval_result: EvaluationResult,
+        pipeline: Pipeline,
+        n_wrong_examples: int = 3,
+        metrics_filter: Optional[Dict[str, List[str]]] = None,
+    ):
+        if any(degree > 1 for node, degree in pipeline.graph.out_degree):
+            logger.warning("Pipelines with junctions are currently not supported.")
+            return
+
+        calculated_metrics = {
+            "": eval_result.calculate_metrics(doc_relevance_col="gold_id_or_answer_match"),
+            "_top_1": eval_result.calculate_metrics(
+                doc_relevance_col="gold_id_or_answer_match", simulated_top_k_reader=1
+            ),
+            " upper bound": eval_result.calculate_metrics(
+                doc_relevance_col="gold_id_or_answer_match", eval_mode="isolated"
+            ),
+        }
+
+        if metrics_filter is not None:
+            for metric_mode in calculated_metrics:
+                calculated_metrics[metric_mode] = {
+                    node: metrics
+                    if node not in metrics_filter
+                    else {metric: value for metric, value in metrics.items() if metric in metrics_filter[node]}
+                    for node, metrics in calculated_metrics[metric_mode].items()
+                }
+
+        pipeline_overview = cls._format_pipeline_overview(calculated_metrics=calculated_metrics, pipeline=pipeline)
+        wrong_samples_report = cls._format_wrong_samples_report(
+            eval_result=eval_result, n_wrong_examples=n_wrong_examples
+        )
+
+        print(f"{pipeline_overview}\n" f"{wrong_samples_report}")
+
+    @classmethod
+    def _format_document_answer(cls, document_or_answer: dict):
+        return "\n \t".join([f"{name}: {value}" for name, value in document_or_answer.items()])
+
+    @classmethod
+    def _format_wrong_sample(cls, query: dict):
```

Comment:
Nitpicking :) :
I would expect this method to be called `_format_wrong_example` as it doesn't format the whole sample, but rather just one example from it.